sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Processor/PartitionLoadBalancer.cs

```diff
@@ -480,9 +480,11 @@ private async Task RenewOwnershipAsync(CancellationToken cancellationToken)
 
                 // Set the EventHubName to null so it doesn't modify the exception message. This exception message is
                 // used so the processor can retrieve the raw Operation string, and adding the EventHubName would append
-                // unwanted info to it.
+                // unwanted info to it. This exception also communicates the PartitionId to the caller.
 
-                throw new EventHubsException(true, null, Resources.OperationClaimOwnership, ex);
+                var exception = new EventHubsException(true, null, Resources.OperationClaimOwnership, EventHubsException.FailureReason.ClaimPartitionFailed, ex);
```

Comment:
Maybe we could do a `SetFailureOperation` and use that as the flag having the user-facing `Reason` stay `General`,  since the operation is being set from a resource, it should be a relatively safe check.

Reply:
Ah, are you suggesting we use the resource as the `Data` key value? That would solve the problem posed in my other comment.

Reply:
I was thinking the resource becomes the value for another well-known key.    Since these are all locally scoped to the exception, having something like:

```
Data :
    Operation :  --RESOURCE THINGER--
    FailureContext:  -- PARTITION--
```

Naming is arbitrary, of course, just to illustrate.   We may be overthinking in terms of avoiding just adding a key for "partition" since it is such a core concept and not something that is esoteric and only makes sense in this particular context.